app/scripts/metamask-controller.js

```diff
@@ -488,13 +484,32 @@ export default class MetamaskController extends EventEmitter {
     });
 
     this.selectedNetworkController = new SelectedNetworkController({
+      state: initState.SelectedNetworkController,
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
         allowedActions: ['NetworkController:getNetworkClientById'],
         allowedEvents: ['NetworkController:stateChange'],
       }),
     });
 
+    if (
+      !this.selectedNetworkController.getNetworkClientIdForDomain('metamask')
+    ) {
+      this.selectedNetworkController.setNetworkClientIdForMetamask(
+        this.networkController.state.selectedNetworkClientId,
+      );
+    }
+
+    let networkClient;
+    if (this.selectedNetworkController.state.perDomainNetwork) {
+      networkClient =
+        this.selectedNetworkController.getProviderAndBlockTracker('metamask');
+    } else {
+      networkClient = this.networkController.getProviderAndBlockTracker();
```

Comment:
These two network clients should be the same, right? They're both proxied clients representing the globally selected network. Is there any functional impact to which of these branches is taken?